lib/okta/directory_service.rb

```diff
@@ -6,20 +6,45 @@ module Okta
   class DirectoryService < Common::Client::Base
     DEFAULT_OKTA_SCOPES = %w[openid profile email address phone offline_access].freeze
 
+    attr_accessor :okta_service
+
+    def initialize
+      @okta_service = Okta::Service.new
+    end
+
     def scopes(category)
-      okta_service = Okta::Service.new
-      servers = okta_service.get_auth_servers
+      # if the category is health we need to call a specific server instead of relying on querying by name,
+      # since there is a 'health/systems' auth server that would affect results
+      category == 'health' ? handle_health_server : handle_nonhealth_server(category)
+    end
+
+    def handle_health_server
+      server = @okta_service.get_auth_server(Settings.directory.health_server_id)
+      scopes = @okta_service.get_server_scopes(server.body['id'])
+      remove_scope_keys(scopes)
+    end
+
+    def handle_nonhealth_server(category)
+      servers = @okta_service.get_auth_servers
       server = servers.body.select { |auth_server| auth_server['name'].include?(category.downcase) }
```

Comment:
maybe handle the use-case if category is nil?
not sure if thats ever possible but even just raising a more specific exception would be better than the null reference exception that would happen on this line